### PR TITLE
Deploying orchestrator based on flist and auto-setup script (closes #784)

### LINKED ISSUE #784: 0-orchestrator flist
# Goal 
Make it extremely easy to get started with 0-orchestrator.

# How & requirements
* Create flist with all the 0-orchestrator dependencies, except for jumpscale (See /docs/setup)
* Then merge this flist with the jumpscale 9 flist and ubuntu to produce the flist people need to use
* Buildscript should be easily configurable for specifying versions of dependencies
* Need to have a very simple way to configure variables when running the flist
  * zerotier network
  * itsyouonline org
  * ...
* You need a working Zero-OS running
* Writing a Go program which use Go client to start a 0-orchestrator container and the differents services needed inside the container (caddy, ays, orchestrator)
* Go program needs to be configurable on the runtime

# Deliverables
* flist buildscript
* cli to start container accepting all the needed variables as commandline arguments
* end-user friendly documentation on how to deploy an orchestrator using the orchestrator flist

### PR DESCRIPTION
**Pull request by maxux**
Here it is ! 
Please check the README.md on this Pull Request.

### CODE REVIEW

File: autosetup/autobootstrap.py [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+                self.progressing()
+                time.sleep(1)
+                continue
+
+            self.progressing(True)
+
+            return ztdata['assignedAddresses'][0].split('/')[0]
+
+    def progress(self):
+        sys.stdout.write("[+] ")
```
Comment (FastGeert) [inline autosetup/autobootstrap.py]: maybe define a specific function to print without newline

File: autosetup/autobootstrap.py [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+        self.tools = OrchestratorInstallerTools()
+
+        self.node = None
+        self.flist = "https://hub.gig.tech/maxux/0-orchestrator-full-autosetup.flist"
+        self.ctname = None
+        self.core_version = "master"
+        self.js_version = "9.1.0"
+        self.templates = "/opt/code/github/zero-os/0-orchestrator/autosetup/templates"
+
+    """
```
Comment (FastGeert) [inline autosetup/autobootstrap.py]: docstrings are put at the first line of the method in python

File: autosetup/autobootstrap.py [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+    def prepare(self, ctname, ztnetwork):
+        self.ctname = ctname
+
+        print("[+] starting orchestrator container")
+        network = [
+            {'type': 'default'},
+            {'type': 'zerotier', 'id': ztnetwork}
+        ]
+
+        """
```
Comment (FastGeert) [inline autosetup/autobootstrap.py]: Shouldn't this be a comment block?

File: autosetup/autobootstrap.py [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+
+        if not self.node.client.filesystem.exists('/var/cache/containers/orchestrator'):
+            self.node.client.filesystem.mkdir('/var/cache/containers/orchestrator')
+
+        cn = self.node.containers.create(
+            name=ctname,
+            flist=self.flist,
+            nics=network,
+            hostname='bootstrap',
+            mounts={'/var/cache/containers/orchestrator': '/optvar'},
```
Comment (FastGeert) [inline autosetup/autobootstrap.py]: This still prevents running more than one orchestrator on one zero-os node, right?

**Comment by rkhamis** [comment]
@maxux please update base branch to alpha-8

**Pull request merged**